File: Easy/Unique Morse Code Words/UniqueMorseCodeWords.js
// https://leetcode.com/problems/unique-morse-code-words

/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function (words) {
  let transformations = new Set();
  let table = [
    ".-",
    "-...",
    "-.-.",
    "-..",
    ".",
    "..-.",
    "--.",
    "....",
    "..",
    ".---",
    "-.-",
    ".-..",
    "--",
    "-.",
    "---",
    ".--.",
    "--.-",
    ".-.",
    "...",
    "-",
    "..-",
    "...-",
    ".--",
    "-..-",
    "-.--",
    "--..",
  ];

  for (let w of words) {
    let t = "";
    for (let c of w) {
      t += table[c.charCodeAt(0) - 97];
    }
    transformations.add(t);
  }

  return transformations.size;
};

File: Easy/Unique Morse Code Words/UniqueMorseCodeWords.cpp
// https://leetcode.com/problems/unique-morse-code-words

class Solution {
public:
  int uniqueMorseRepresentations(vector<string> &words) {
    ios::sync_with_stdio(0), cin.tie(0), cin.tie(0);

    vector<string> table{".-",   "-...", "-.-.", "-..",  ".",    "..-.", "--.",
                         "....", "..",   ".---", "-.-",  ".-..", "--",   "-.",
                         "---",  ".--.", "--.-", ".-.",  "...",  "-",    "..-",
                         "...-", ".--",  "-..-", "-.--", "--.."};

    int size = words.size();
    unordered_set<string> transformations;
    for (int i = 0; i < size; i++) {
      string t = "";
      for (char c : words[i]) {
        t += table[c - 'a'];
      }
      transformations.insert(t);
    }
    return transformations.size();
  }
};